packages/client/src/components/SafeSettings.js

```diff
@@ -467,7 +516,31 @@ function SafeSettings({ address, web3, name, threshold, safeOwners }) {
   const onRemoveSafeOwnerSubmit = async (ownerToBeRemoved) => {
 
     if (ownerToBeRemoved) {
-      await proposeRemoveSigner(formatAddress(ownerToBeRemoved.address));
+      const latestBlock = await web3.injectedProvider
+        .send([web3.injectedProvider.getBlock(true)])
+        .then(web3.injectedProvider.decode);
+
+      const { height, id } = latestBlock;
+      const ownerToBeRemovedHex = Buffer.from(`Remove ${ownerToBeRemoved.address} as a signer.`).toString("hex");
+
+      const message = `${ownerToBeRemovedHex}${id}`;
+      const messageHex = Buffer.from(message).toString("hex");
+
+      let sigResponse = await web3.injectedProvider
+        .currentUser()
+        .signUserMessage(messageHex);
+      const sigMessage =
+        sigResponse[0]?.signature?.signature ?? sigResponse[0]?.signature;
+      const keyIds = [sigResponse[0]?.keyId];
+      const signatures = [sigMessage];
+
+      await proposeRemoveSigner(
+        formatAddress(ownerToBeRemoved.address),
+        message,
+        keyIds,
+        signatures,
+        height
+      );
```

Comment:
Since this part of the logic appears 4 times in our code, can we extract it into a utility function?

Reply:
Yeah, makes sense, a significant part is fixed so it can be moved to utility

Reply:
Moved to util